Improve testing on Travis

0xced opened the pull request:
* Use latest OS X image (Xcode 7)
* Run tests on multiple iOS versions (8.1, 8.2, 8,3, 8.4 and 9.0)
* Use xcpretty for a readable output
* Use gstdbuf for live output of test results

phatmann commented:
Our Travis definitely needs an update, so I am happy to see this PR. However, I have a few questions:

* Why would we run the tests on all the flavors of iOS 8? Just 8.4 will do.
* Where is the test for iOS 9 configured?

0xced commented:
I removed iOS 8.1 to 8.3 and added iOS 9.0. Unfortunately, Travis is currently experiencing problems with the Xcode 7 image. There’s travis-ci/travis-ci#3040 which is tracking this issue.

phatmann commented:
@0xced, do you want me to sit on this PR until the Travis issue is resolved?

0xced commented:
Actually, I just found a workaround and the tests are now running on Travis with Xcode 7, yay! They are all green ✅ on iOS 8.4 but [two tests are failing](https://travis-ci.org/kif-framework/KIF/jobs/84399020#L601-L602) ❌ on iOS 9.0.

phatmann commented:
Excellent work!
